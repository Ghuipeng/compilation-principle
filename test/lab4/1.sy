int a;

int main()
{
    int abb,bcc;
    a = 1 + 2;
    a = 0777;
    a = 0x777;
    a = 0X777;
    // fffffffff
    float b = 0.3e+0;
    b = 0.3e+1f;
    if(a < 5)
        return 1;
    return 0;
}
