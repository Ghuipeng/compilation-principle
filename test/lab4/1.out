     token	     lexme	lineno	     scope	     value
       INT	       int	     1
        ID	         a	     1	         1	         1
 SEMICOLON	         ;	     1
       INT	       int	     3
        ID	      main	     3	         1	         3
    LPAREN	         (	     3
    RPAREN	         )	     3
    LBRACE	         {	     4
       INT	       int	     5
        ID	       abb	     5	       1-1	         5
     COMMA	         ,	     5
        ID	       bcc	     5	       1-1	         5
 SEMICOLON	         ;	     5
        ID	         a	     6	         1	         1
    ASSIGN	         =	     6
   DECIMAL	         1	     6	          	         1
       ADD	         +	     6
   DECIMAL	         2	     6	          	         2
 SEMICOLON	         ;	     6
        ID	         a	     7	         1	         1
    ASSIGN	         =	     7
     OCTAL	      0777	     7	          	       511
 SEMICOLON	         ;	     7
        ID	         a	     8	         1	         1
    ASSIGN	         =	     8
HEXADECIMAL	     0x777	     8	          	      1911
 SEMICOLON	         ;	     8
        ID	         a	     9	         1	         1
    ASSIGN	         =	     9
HEXADECIMAL	     0X777	     9	          	      1911
 SEMICOLON	         ;	     9
     FLOAT	     float	    11
        ID	         b	    11	       1-1	        11
    ASSIGN	         =	    11
  FLOATNUM	    0.3e+0	    11	          	 0.3000000
 SEMICOLON	         ;	    11
        ID	         b	    12	       1-1	        11
    ASSIGN	         =	    12
  FLOATNUM	   0.3e+1f	    12	          	 3.0000000
 SEMICOLON	         ;	    12
        IF	        if	    13
    LPAREN	         (	    13
        ID	         a	    13	         1	         1
      LESS	         <	    13
   DECIMAL	         5	    13	          	         5
    RPAREN	         )	    13
    RETURN	    return	    14
   DECIMAL	         1	    14	          	         1
 SEMICOLON	         ;	    14
    RETURN	    return	    15
   DECIMAL	         0	    15	          	         0
 SEMICOLON	         ;	    15
    RBRACE	         }	    16
